### ISSUE TITLE
Don't try to read imported JSON files

### ISSUE THREAD

**Issue by sindresorhus** [benmosher/eslint-plugin-import#267]
In Node.js, this is valid and a common pattern:

```js
var pkg = require('./package.json');
```

I got errors when I ran this plugin on [`got`](https://github.com/sindresorhus/got):

```
  test/headers.js:3
  ✖  3:17  Parse errors in imported module '../package.json': Line 2: Unexpected token (2:9)  import/namespace
  ✖  3:17  Parse errors in imported module '../package.json': Line 2: Unexpected token (2:9)  import/no-named-as-default-member
  ✖  3:17  Parse errors in imported module '../package.json': Line 2: Unexpected token (2:9)  import/no-named-as-default
```

**Comment by benmosher** [comment]
You got those errors on a `require`?

**Comment by sindresorhus** [comment]
No, here's the relevant statement: https://github.com/sindresorhus/got/blob/67ee190881f4ba9c498708dc41c1d71c5b6039a2/test/headers.js#L3

**Comment by benmosher** [comment]
Ah, okay, that makes sense.

I think it probably makes sense to handle this in the Node resolver and return `null` so the analysis is ignored. (analogous to defining `\.json$` as an ignored pattern)

I am assuming you're not expecting it to attempt to lint a `json` file's imports.

**Comment by sindresorhus** [comment]
> I am assuming you're not expecting it to attempt to lint a json file's imports.

Correct.

**Comment by benmosher** [comment]
Actually, I suddenly think it would make more sense to have a parseable extension whitelist (`import/module-extensions` setting?) and default to `['js', 'mjs']`. That would make #268 really efficient in the case that a project is actually using `.mjs` everywhere; could set the extensions to just `['mjs']` and it wouldn't even need to read the file and inspect the contents.

Might make sense to also make a rule to flag imports of files that don't match that list, I would want some way to avoid having accidental `import`s of CoffeeScript or something silently appear valid.

**Comment by sindresorhus** [comment]
I like the idea of such option, but should default to `['js']` for now. `mjs` is far from decided on.

**Comment by benmosher** [comment]
Sure. Maybe a better expression of the default would have been `['js', BIKESHEDDED_MODULE_EXT]`.

**Comment by benmosher** [comment]
Bumped out of `next` milestone pending discussion on #297.
